WalletWasabi/Blockchain/Transactions/TransactionStore.cs

```diff
@@ -1,411 +1,327 @@
+using Microsoft.Data.Sqlite;
 using NBitcoin;
-using Nito.AsyncEx;
 using System.Collections.Generic;
+using System.Diagnostics;
 using System.Diagnostics.CodeAnalysis;
 using System.IO;
 using System.Linq;
+using System.Text;
 using System.Threading;
 using System.Threading.Tasks;
-using WalletWasabi.Blockchain.Transactions.Operations;
 using WalletWasabi.Extensions;
 using WalletWasabi.Helpers;
 using WalletWasabi.Io;
 using WalletWasabi.Logging;
-using WalletWasabi.Nito.AsyncEx;
+using WalletWasabi.Stores;
 
 namespace WalletWasabi.Blockchain.Transactions;
 
 public class TransactionStore : IAsyncDisposable
 {
-	public TransactionStore(string workFolderPath, Network network)
+	public TransactionStore(string workFolderPath, Network network, bool migrateData = true)
 	{
-		WorkFolderPath = Guard.NotNullOrEmptyOrWhitespace(nameof(workFolderPath), workFolderPath, trim: true);
-		Network = network;
+		workFolderPath = Guard.NotNullOrEmptyOrWhitespace(nameof(workFolderPath), workFolderPath, trim: true);
 
-		// In Transactions.dat every line starts with the tx id, so the first character is the best for digest creation.
-		TransactionsFileManager = new IoManager(filePath: Path.Combine(WorkFolderPath, "Transactions.dat"));
+		if (workFolderPath == SqliteStorageHelper.InMemoryDatabase)
+		{
+			DataSource = SqliteStorageHelper.InMemoryDatabase;
+		}
+		else
+		{
+			IoHelpers.EnsureDirectoryExists(workFolderPath);
+			DataSource = Path.Combine(workFolderPath, "Transactions.sqlite");
+
+			// TODO: Remove. Useful for testing.
+			// if (File.Exists(dataSource))
+			// {
+			// 	 File.Delete(dataSource);
+			// }
+		}
+
+		SqliteStorage = TransactionSqliteStorage.FromFile(dataSource: DataSource, network);
+
+		if (migrateData)
+		{
+			string oldPath = Path.Combine(workFolderPath, "Transactions.dat");
+			Import(oldPath, network, deleteAfterImport: false);
```

Comment:
In the final version, the old file will be deleted after import?

Reply:
I agree to create a PR that deletes the file after this PR is merged - better for the devs they can switch back and forth to other PR-s where the database does not exist without losing Tx.dat. Drop the idea of keeping the tx.dat on the disk, we do not need it.